tutorials/create-serverless-scraping/index.mdx

```diff
@@ -0,0 +1,361 @@
+
+    SQS credentials and queue URL are read by the function from environment variables. Those variables are set by Terraform as explained in [one of the next sections](#create-a-terraform-file-to-provision-the-necessary-scw-resources). *If you choose another deployment method (e.g: console) don't forget to set them.*
+    ```python
+    queue_url = os.getenv('QUEUE_URL') 
+    sqs_access_key = os.getenv('SQS_ACCESS_KEY')
+    sqs_secret_access_key = os.getenv('SQS_SECRET_ACCESS_KEY')
+    ```
+    Use the `requests` package to retrieve the HTML, and `BeautifulSoup` to parse it and extract the `span` we are interested in: `titleline` containing **title** and **url**, and `age` holding the publication time 
+    ```python
+    def scrape_and_push():
+        page = requests.get(HN_URL)
+        html_doc = page.content
+
+        soup = BeautifulSoup(html_doc, 'html.parser')
+
+        titlelines = soup.find_all(class_="titleline")
+        ages = soup.find_all(class_="age")
+    ```
+    Using the AWS python sdk `boto3`, connect to the SQS queue and push `title` and `url` of articles published less than 15 minutes ago.
+    ```python
+        sqs = boto3.client(
+          'sqs', 
+          endpoint_url=SCW_SQS_URL, 
+          aws_access_key_id=sqs_access_key, 
+          aws_secret_access_key=sqs_secret_access_key, 
+          region_name='fr-par')
+
+        for age, titleline in zip(ages, titlelines):
+            time_str = age["title"]
+            time = datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S")
+            if datetime.utcnow() - time > timedelta(minutes=15):
+                continue
+
+            body = json.dumps({'url': titleline.a["href"], 'title': titleline.a.get_text()})
+            response = sqs.send_message(QueueUrl=queue_url, MessageBody=body)
+    ```
+5. As explained in the [Scaleway Functions documentation,](https://www.scaleway.com/en/docs/serverless/functions/how-to/package-function-dependencies-in-zip/#how-to-configure-your-package-using-python), install the dependencies in a folder `package` and create the archive to be uploaded:
+    ```bash
+    cd scraper
+    pip install -r requirements.txt --target ./package
+    zip -r functions.zip handlers/ package/
+    ```
+
+## Create the consumer
+Next, let's create our consumer function. When receiving a message containing the article's title and URL from the queue, it will scrape the page for some stats (number of `a`, `h1` and `p` tags) and save these values in a Scaleway Managed PostgreSQL Database instance. We show how to create the database instance (and other underlying resources) via Terraform in the following step. If you opt for another method to create the underlying resources, e.g. the console or API, don't forget to set the relevant function environment variables.
```

Comment:
```suggestion
Next, let's create our consumer function. When receiving a message containing the article's title and URL from the queue, it will scrape the page for some stats (number of `a`, `h1` and `p` tags) and save these values in a Scaleway Managed PostgreSQL Database instance. We show how to create the database instance (and other underlying resources) via Terraform in the following steps. If you opt for another method to create the underlying resources, e.g. the console or API, do not forget to set the relevant function environment variables.
```